src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -63,12 +64,20 @@ public WebProxy(string? Address, bool BypassOnLocal, string[]? BypassList, ICred
 
         public bool BypassProxyOnLocal { get; set; }
 
+        [AllowNull]
         public string[] BypassList
         {
             get { return _bypassList != null ? (string[])_bypassList.ToArray(typeof(string)) : Array.Empty<string>(); }
             set
             {
-                _bypassList = new ArrayList(value);
+                if (value == null)
+                {
+                    _bypassList = null;
+                }
+                else
+                {
+                    _bypassList = new ArrayList(value);
+                }
```

Comment:
Nit: I'd suggest writing this as:
```suggestion
                _bypassList = value != null ? new ArrayList(value) : null;
```